tron/core/actiongraph.py

```diff
@@ -61,10 +61,20 @@ def get_required_actions(self, name):
         """
         if name not in self.action_map:
             return []
-        return self.action_map[name].required_actions
+
+        return (
+            self.action_map[action]
+            for action in self.action_map[name].required_actions
+        )
 
     def get_dependent_actions(self, name):
-        return self.action_map[name].dependent_actions
+        if name not in self.action_map:
```

Comment:
Can you help me understand why this funtion needs to change in this new world?

Is it because we can't return a set? It has a default of an empty set, does that not work or do we get a key error? I don't understand.

Reply:
not entirely sure which change you're asking about so i'll address both:

- the `if` change isn't strictly necessary, i just copied it from `get_required_actions` because i like consistency

- the change in `return` statement is necessary because we need to return `Action` objects but we no longer have them in `required_actions / dependent_actions` - we only have action names